src/main/java/gtPlusPlus/xmod/gregtech/common/tileentities/machines/multi/production/GregtechMetaTileEntity_SolarTower.java

```diff
@@ -610,21 +608,13 @@ public boolean checkRecipe(final ItemStack aStack) {
 			ArrayList<FluidStack> aFluids = this.getStoredFluids();
 			for (FluidStack aFluid : aFluids) {
 				if (aFluid.getFluid().equals(mColdSalt)) {
-					int aFluidAmount = aFluid.amount;
-					if (aFluidAmount > this.mHeatLevel) {
-						this.mHeatLevel = 0;
-						log("Removed all Heat");
-						this.depleteInput(FluidUtils.getFluidStack(mColdSalt, this.mHeatLevel));
-						this.addOutput(FluidUtils.getFluidStack(mHotSalt, this.mHeatLevel));
-
-					}
-					else {
-						this.mHeatLevel -= aFluidAmount;
-						log("Removed Heat: " + (aFluidAmount));
-						this.depleteInput(FluidUtils.getFluidStack(mColdSalt, aFluidAmount));
-						this.addOutput(FluidUtils.getFluidStack(mHotSalt, aFluidAmount));
-
-					}
+					int aFluidAmount = Math.min(aFluid.amount, this.mHeatLevel);
+
+					this.mHeatLevel -= aFluidAmount;
+					log("Removed Heat: " + (aFluidAmount));
+					this.depleteInput(FluidUtils.getFluidStack(mColdSalt, aFluidAmount));
+					this.addOutput(FluidUtils.getFluidStack(mHotSalt, aFluidAmount));
+
 					break;
 				}
 			}			
```

Comment:
Add `Math.max(mHeatLevel, 0) at the end of this method to prevent negative heat

Reply:
Done, although I don't know how heat would be negative after Math,min(aFluid.amount, this.mHeatLevel).